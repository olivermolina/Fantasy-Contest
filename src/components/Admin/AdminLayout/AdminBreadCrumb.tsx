import React from 'react';
import Link from '@mui/material/Link';
import HomeIcon from '@mui/icons-material/Home';
import NavigateNextIcon from '@mui/icons-material/NavigateNext';
import Breadcrumbs from '@mui/material/Breadcrumbs';
import { useRouter } from 'next/router';
import { UrlPaths } from '~/constants/UrlPaths';

const breadcrumbNameMap: { [key: string]: string } = {
  [UrlPaths.AdminOffer]: 'Offers',
  [UrlPaths.AdminPendingBets]: 'Settle Pending Picks',
  [UrlPaths.AdminUsersBalance]: 'Users Balance',
  [UrlPaths.AdminWeeklyBalance]: 'Weekly Balance',
  [UrlPaths.AdminPlayFreeBonusCredit]: 'Play Free Bonus Credit',
  [UrlPaths.AdminWithdrawable]: 'Manage Amount Available to Withdraw',
  [UrlPaths.AdminLineExposure]: 'Line Exposure',
  [UrlPaths.ManageWithdrawalOffer]: 'Manage Withdrawal Offer',
  [UrlPaths.SendSMS]: 'Send SMS',
  [UrlPaths.ManageAgentReferralCodes]: 'Manage Partner Referral Codes',
  [UrlPaths.PartnersReferralCodes]: 'Manage Partner Referral Codes',
  [UrlPaths.ManagePartnersPams]: 'Add/Edit Agents & PAMs',
  [UrlPaths.ManageUsers]: 'Add/Edit Users',
  [UrlPaths.AdminUserLimits]: 'Entry Limits',
};

export default function AdminBreadCrumb() {
  const { pathname } = useRouter();

  return (
    <Breadcrumbs
      aria-label="breadcrumb"
      className={'bg-gray-200 px-6 py-2'}
      separator={<NavigateNextIcon fontSize="small" />}
    >
      <Link
        underline="hover"
        sx={{ display: 'flex', alignItems: 'center' }}
        color="inherit"
        href="/admin"
      >
        <HomeIcon sx={{ mr: 0.5 }} fontSize="inherit" />
        Home
      </Link>
      {pathname !== UrlPaths.Admin && pathname !== UrlPaths.Partners && (
        <Link
          underline="hover"
          sx={{ display: 'flex', alignItems: 'center' }}
          color="inherit"
          href={pathname}
        >
          {breadcrumbNameMap[pathname] ||
            pathname
              .replace('/admin/', '')
              .replace('/partners/', '')
              .replace(/\b[a-z]/g, (x) => x.toUpperCase())
              .replaceAll('-', ' ')}
        </Link>
      )}
    </Breadcrumbs>
  );
}
