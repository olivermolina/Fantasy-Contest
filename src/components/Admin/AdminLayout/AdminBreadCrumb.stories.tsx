import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';

import AdminBreadCrumb from './AdminBreadCrumb';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Lockspread/Admin',
  component: AdminBreadCrumb,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    backgroundColor: { control: 'color' },
  },
} as ComponentMeta<typeof AdminBreadCrumb>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof AdminBreadCrumb> = () => (
  <AdminBreadCrumb />
);

export const BreadCrumb = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args

BreadCrumb.args = {};
