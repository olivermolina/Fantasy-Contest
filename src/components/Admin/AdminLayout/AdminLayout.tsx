import * as React from 'react';
import { useEffect } from 'react';
import { CSSObject, styled, Theme, useTheme } from '@mui/material/styles';
import Box from '@mui/material/Box';
import MuiDrawer from '@mui/material/Drawer';
import MuiAppBar, { AppBarProps as MuiAppBarProps } from '@mui/material/AppBar';
import Toolbar from '@mui/material/Toolbar';
import List from '@mui/material/List';
import CssBaseline from '@mui/material/CssBaseline';
import Divider from '@mui/material/Divider';
import IconButton from '@mui/material/IconButton';
import MenuIcon from '@mui/icons-material/Menu';
import ChevronLeftIcon from '@mui/icons-material/ChevronLeft';
import ChevronRightIcon from '@mui/icons-material/ChevronRight';
import ListItem from '@mui/material/ListItem';
import ListItemButton from '@mui/material/ListItemButton';
import ListItemIcon from '@mui/material/ListItemIcon';
import ListItemText from '@mui/material/ListItemText';
import TrendingUpIcon from '@mui/icons-material/TrendingUp';

import { Logo } from '~/components/Layout/Logo';
import { Tooltip } from '@mui/material';
import { UrlPaths } from '~/constants/UrlPaths';
import AdminBreadCrumb from '~/components/Admin/AdminLayout/AdminBreadCrumb';
import { NextRouter } from 'next/router';
import LogoutIcon from '@mui/icons-material/Logout';
import { User } from '@prisma/client';
import Avatar from '@mui/material/Avatar';
import Header from '../../Header/Header';

const drawerWidth = 180;

const openedMixin = (theme: Theme): CSSObject => ({
  width: drawerWidth,
  transition: theme.transitions.create('width', {
    easing: theme.transitions.easing.sharp,
    duration: theme.transitions.duration.enteringScreen,
  }),
  overflowX: 'hidden',
});

const closedMixin = (theme: Theme): CSSObject => ({
  transition: theme.transitions.create('width', {
    easing: theme.transitions.easing.sharp,
    duration: theme.transitions.duration.leavingScreen,
  }),
  overflowX: 'hidden',
  width: 0,
});

const DrawerHeader = styled('div')(({ theme }) => ({
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'space-between',
  padding: theme.spacing(0, 1),
  // necessary for content to be below app bar
  ...theme.mixins.toolbar,
  backgroundColor: '#2562ea',
}));

interface AppBarProps extends MuiAppBarProps {
  open?: boolean;
}

const AppBar = styled(MuiAppBar, {
  shouldForwardProp: (prop) => prop !== 'open',
})<AppBarProps>(({ theme, open }) => ({
  zIndex: theme.zIndex.drawer + 1,
  transition: theme.transitions.create(['width', 'margin'], {
    easing: theme.transitions.easing.sharp,
    duration: theme.transitions.duration.leavingScreen,
  }),
  ...(open && {
    marginLeft: drawerWidth,
    width: `calc(100% - ${drawerWidth}px)`,
    transition: theme.transitions.create(['width', 'margin'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen,
    }),
  }),
  minHeight: '90px',
}));

const Drawer = styled(MuiDrawer, {
  shouldForwardProp: (prop) => prop !== 'open',
})(({ theme, open }) => ({
  width: drawerWidth,
  flexShrink: 0,
  whiteSpace: 'nowrap',
  boxSizing: 'border-box',
  ...(open && {
    ...openedMixin(theme),
    '& .MuiDrawer-paper': openedMixin(theme),
  }),
  ...(!open && {
    ...closedMixin(theme),
    '& .MuiDrawer-paper': closedMixin(theme),
  }),
}));

const MENU_OPTIONS = [
  {
    id: 'challenge',
    label: 'Challenge',
    path: UrlPaths.Challenge,
    icon: <TrendingUpIcon />,
  },
  {
    id: 'logout',
    label: 'Logout',
    path: UrlPaths.Logout,
    icon: <LogoutIcon />,
  },
];

interface AdminLayoutProps {
  router: NextRouter;
  onMenuItemClick: (path: string) => void;
  children?: JSX.Element | React.ReactNode;
  user?: User;
}

export default function AdminLayout(props: AdminLayoutProps) {
  const { router, onMenuItemClick, user } = props;
  const theme = useTheme();
  const [open, setOpen] = React.useState(false);
  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
  const menuOpen = Boolean(anchorEl);
  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
    setAnchorEl(event.currentTarget);
  };

  const handleDrawerOpen = () => {
    setOpen(true);
  };

  const handleDrawerClose = () => {
    setOpen(false);
  };

  useEffect(() => {
    window.scrollTo(0, 0);
  }, [router.pathname]);

  React.useEffect(() => {
    const resizeOps = () => {
      document.documentElement.style.setProperty(
        '--vh',
        window.innerHeight * 0.01 + 'px',
      );
    };
    resizeOps();
    window.addEventListener('resize', resizeOps);
    document.getElementById('main')?.scrollIntoView();
  }, []);

  return (
    <>
      <Header>
        <meta
          name="viewport"
          content="width=1450, height=1600, initial-scale=1"
        />
      </Header>
      <Box sx={{ display: 'flex' }}>
        <CssBaseline />
        <AppBar position="fixed" open={open}>
          <Toolbar
            sx={{
              height: '90px',
              backgroundColor: '#2562ea',
            }}
          >
            <IconButton
              color="inherit"
              aria-label="open drawer"
              onClick={handleDrawerOpen}
              edge="start"
              sx={{
                marginRight: 2,
                ...(open && { display: 'none' }),
              }}
            >
              <MenuIcon />
            </IconButton>
            {!open && <Logo key="admin-top" isLoggedIn={true} />}
            <span className={'flex-grow'} />

            <Tooltip title={user?.username?.toUpperCase()}>
              <IconButton
                onClick={handleClick}
                size="small"
                sx={{ ml: 2 }}
                aria-controls={menuOpen ? 'account-menu' : undefined}
                aria-haspopup="true"
                aria-expanded={menuOpen ? 'true' : undefined}
              >
                <Avatar sx={{ width: 50, height: 50 }}>
                  {user?.username?.charAt(0).toUpperCase()}
                </Avatar>
              </IconButton>
            </Tooltip>
          </Toolbar>
          <AdminBreadCrumb />
        </AppBar>
        <Drawer variant="permanent" open={open}>
          <DrawerHeader>
            <span />
            <Logo key="top" isLoggedIn={true} />
            <IconButton onClick={handleDrawerClose}>
              {theme.direction === 'rtl' ? (
                <ChevronRightIcon sx={{ color: 'white' }} />
              ) : (
                <ChevronLeftIcon sx={{ color: 'white' }} />
              )}
            </IconButton>
          </DrawerHeader>
          <Divider />
          <List>
            {MENU_OPTIONS.map((option) => (
              <ListItem
                key={option.id}
                disablePadding
                sx={{ display: 'block' }}
              >
                <ListItemButton
                  sx={{
                    minHeight: 48,
                    justifyContent: open ? 'initial' : 'center',
                    px: 2.5,
                  }}
                  onClick={() => onMenuItemClick(option.path)}
                  selected={option.path.toString() === router?.pathname}
                >
                  <ListItemIcon
                    sx={{
                      minWidth: 0,
                      mr: open ? 3 : 'auto',
                      justifyContent: 'center',
                    }}
                  >
                    {option.icon}
                  </ListItemIcon>
                  <ListItemText
                    primary={option.label}
                    sx={{ opacity: open ? 1 : 0 }}
                  />
                </ListItemButton>
              </ListItem>
            ))}
          </List>
          <Divider />
        </Drawer>
        <Box
          component="main"
          sx={{ flexGrow: 1, overflow: 'hidden' }}
          id={'main'}
        >
          <DrawerHeader />
          <div
            className={
              'mt-14 p-5 min-h-screen w-full overflow-y-hidden overflow-x-auto block'
            }
          >
            {props.children}
          </div>
        </Box>
      </Box>
    </>
  );
}
