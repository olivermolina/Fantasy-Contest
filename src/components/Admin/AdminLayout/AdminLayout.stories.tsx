import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';

import AdminLayout from './AdminLayout';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Lockspread/Admin',
  component: AdminLayout,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    backgroundColor: { control: 'color' },
  },
} as ComponentMeta<typeof AdminLayout>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof AdminLayout> = (args) => (
  <AdminLayout {...args} />
);

export const Layout = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args

Layout.args = {};
