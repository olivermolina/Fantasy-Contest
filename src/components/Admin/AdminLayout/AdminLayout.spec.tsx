import React from 'react';
import { render } from '@testing-library/react';
import AdminLayout from './AdminLayout';
import { useRouter } from 'next/router';

window.HTMLElement.prototype.scrollIntoView = jest.fn;
jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '/',
      pathname: '',
      query: '',
      asPath: '',
    };
  },
}));
describe('AdminLayout', () => {
  it('should render', () => {
    const router = useRouter();
    const { container } = render(
      <AdminLayout
        router={router}
        onMenuItemClick={() => console.log('click')}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
