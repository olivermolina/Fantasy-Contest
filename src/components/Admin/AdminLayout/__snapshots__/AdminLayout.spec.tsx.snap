// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminLayout should render 1`] = `
<div>
  <div
    class="MuiBox-root css-k008qs"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1ffy8ox-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-panklt-MuiToolbar-root"
      >
        <button
          aria-label="open drawer"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-134qg7o-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="MenuIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <a
          href="/challenge"
          rel="noreferrer"
          target="_self"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2770%27%20height=%2790%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="LockSpread"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </a>
        <span
          class="flex-grow"
        />
        <button
          aria-haspopup="true"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-137ul40-MuiButtonBase-root-MuiIconButton-root"
          data-mui-internal-clone-element="true"
          tabindex="0"
          type="button"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-zgt3rr-MuiAvatar-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
              data-testid="PersonIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <nav
        aria-label="breadcrumb"
        class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root bg-gray-200 px-6 py-2 css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
      >
        <ol
          class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
        >
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-1f2xi7l-MuiTypography-root-MuiLink-root"
              href="/admin"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vnn4cn-MuiSvgIcon-root"
                data-testid="HomeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
              Home
            </a>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
              data-testid="NavigateNextIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-1f2xi7l-MuiTypography-root-MuiLink-root"
              href=""
            />
          </li>
        </ol>
      </nav>
    </header>
    <div
      class="MuiDrawer-root MuiDrawer-docked css-1nk3zll-MuiDrawer-docked"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-12i7wg6-MuiPaper-root-MuiDrawer-paper"
      >
        <div
          class="css-103d7q3"
        >
          <span />
          <a
            href="/challenge"
            rel="noreferrer"
            target="_self"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2770%27%20height=%2790%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="LockSpread"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </a>
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1d159sf-MuiSvgIcon-root"
              data-testid="ChevronLeftIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
        <ul
          class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
        >
          <li
            class="MuiListItem-root MuiListItem-gutters css-uxa1ks-MuiListItem-root"
          >
            <div
              class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1fqi8hc-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemIcon-root css-1ddsiuw-MuiListItemIcon-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="TrendingUpIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m16 6 2.29 2.29-4.88 4.88-4-4L2 16.59 3.41 18l6-6 4 4 6.3-6.29L22 12V6z"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root css-1cag644-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  Challenge
                </span>
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters css-uxa1ks-MuiListItem-root"
          >
            <div
              class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1fqi8hc-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemIcon-root css-1ddsiuw-MuiListItemIcon-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="LogoutIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m17 7-1.41 1.41L18.17 11H8v2h10.17l-2.58 2.58L17 17l5-5zM4 5h8V3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h8v-2H4V5z"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root css-1cag644-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  Logout
                </span>
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </div>
          </li>
        </ul>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
      </div>
    </div>
    <main
      class="MuiBox-root css-ndu8ol"
      id="main"
    >
      <div
        class="css-103d7q3"
      />
      <div
        class="mt-14 p-5 min-h-screen w-full overflow-y-hidden overflow-x-auto block"
      />
    </main>
  </div>
</div>
`;
